main.go

```diff
@@ -16,6 +16,7 @@ import (
 
 var (
 	kubeconfig string
+	serverIP   string
```

Comment:
I reviewed the entire path of this parameter and came to the conclusion that it does not need to be set separately. If the kubconfig is set, this should already be considered as a out-of-cluster mode: IP of the API server can be obtained from kubeconfig.
1. See TlsCertificateGeneraterequest: the addition of serverIP to the alt names of the certificate is already implemented, so the user should not generate his own certificate: the setting of kubeconfig in parameters should be enough. This function sets CommonName as Service name according to the field Service in TlsCertificateProps structure. This structure is created in GetTLSCertProps, so you should change the CN there by adding some flag to the Client or to the function GetTLSCertProps.
2. webhooks.NewWebhookRegistrationClient takes the IP, but it should be extracted from kubeconfig. To construct webhook configs for debug, please use flag "whether kubeconfig was passed to the main", i.e. "out-of-cluster" flag.
3. serverIP should not be saved neither in WebhookRegistrationClient nor in Client, because it's a temporary parameter which is needed only on start.

Reply:
In my case, my localhost(10.0.0.1) and the instance where apiserver(10.0.0.2) is running are sharing the same network, when starting the Kyverno out of the cluster(run it on 10.0.0.1), the url in the `MutatingWebhookConfiguration` should be the ip of localhost so that the apiserver knows where to get the `/mutate` and `/validate`.
This serverIP holds the ip of the machine to run the controller, it is used to register mutatation and validation webhooks.